File: src/Website/Models/Resource/StatsPageResult.cs
using Newtonsoft.Json;
using Website.Models.Database;

namespace Website.Models.Resource
{
    public class StatsPageResult
    {
        [JsonProperty("history")]
        public ChartObject? History { get; set; }

        [JsonProperty("found_at_stats")]
        public ChartObject? FoundAtStats { get; set; }

        [JsonProperty("character_stats")]
        public ChartObject? CharacterStats { get; set; }

        [JsonProperty("curse_stats")]
        public ChartObject? CurseStats { get; set; }

        [JsonProperty("floor_stats")]
        public ChartObject? FloorStats { get; set; }

        [JsonProperty("transformation_item_ranking")]
        public ChartObject? TransformationItemRanking { get; set; }

        [JsonProperty("videos")]
        public NldbVideoResult? Videos { get; set; }
    }
}

File: src/Website/Ts/Models/StatsPageResult.ts
import { ChartData } from 'chart.js'
import { VideoResult } from './video-result';
import { IsaacResource } from './isaac-resource';

export interface StatsPageResult {
    history?: ChartData;
    found_at_stats?: ChartData;
    character_stats?: ChartData;
    curse_stats?: ChartData;
    floor_stats?: ChartData;
    transformation_item_ranking?: ChartData;
    videos?: VideoResult;
    resource: IsaacResource;
}

